src/api/sso/test/row-level-policies.test.js

```diff
@@ -0,0 +1,350 @@
+      const { status } = await supabase.from('feeds').delete().match({
+        url: feedList[0].url,
+      });
+      expect(status).toBe(200);
+      await ensureFeeds(feedList);
+    });
+  });
+});
+
+describe('Authenticated users through Seneca SSO', () => {
+  beforeAll(async () => {
+    const { email, firstName, lastName, displayName, githubAvatarUrl } = hansLippershey;
+    const accessToken = createToken(
+      email,
+      firstName,
+      lastName,
+      displayName,
+      ['seneca', 'telescope'],
+      githubAvatarUrl
+    );
+    // overwrite public anon_key with a JWT sent from the server
+    supabase.auth.setAuth(accessToken);
+  });
+
+  afterAll(async () => {
+    supabase.auth.setAuth(ANON_KEY);
+  });
+
+  describe('"telescope_profiles" security policies', () => {
+    beforeEach(async () => {
+      await createTelescopeUsers(users);
+    });
+
+    afterEach(async () => {
+      await cleanupTelescopeUsers(users);
+    });
+
+    it('should return all users', async () => {
+      const { status, data: profiles } = await supabase.from('telescope_profiles').select('*');
+
+      expect(status).toBe(200);
+      expect(profiles.length).toBe(users.length);
+    });
+
+    it('should not allow creating new users', async () => {
+      const { status } = await supabase.from('telescope_profiles').insert(
+        {
+          id: hash(unknowUser.email),
+          email: unknowUser.email,
+          first_name: unknowUser.firstName,
+          last_name: unknowUser.lastName,
+          display_name: unknowUser.displayName,
+          github_username: unknowUser.githubUsername,
+          github_avatar_url: unknowUser.githubAvatarUrl,
+        },
+        { returning: 'minimal' }
+      );
+
+      expect(status).toBe(401);
+    });
+
+    it('should allow users to update their own profiles', async () => {
+      const { status, data: profiles } = await supabase
+        .from('telescope_profiles')
+        .update({ display_name: unknowUser.displayName }, { returning: 'representation' })
+        .match({ id: hash(hansLippershey.email) });
+
+      expect(status).toBe(200);
+      expect(profiles.length).toBe(1);
+      expect(profiles[0].display_name).toBe(unknowUser.displayName);
+    });
+
+    it("should not allow users to update others' profiles", async () => {
+      const { status } = await supabase
+        .from('telescope_profiles')
+        .update({ display_name: unknowUser.displayName })
+        .match({ id: hash(johannesKepler.email) });
+
+      expect(status).toBe(404);
+    });
+
+    it('should not allow deleting users', async () => {
+      const { status } = await supabase
+        .from('telescope_profiles')
+        .delete()
+        .match({
+          id: hash(hansLippershey.email),
+        });
+
+      expect(status).toBe(200);
+      await ensureUsers(users);
+    });
+  });
+
+  describe('"feeds" security policies', () => {
+    // Create feeds for HansLippershey and an unknown user
```

Comment:
Who is the "unknown user"?

Reply:
I mean an unregistered user, I'll change it